packages/sanity/src/structure/panes/document/timeline/events/PublishedEventMenu.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  Flex,
+  Menu,
+  // eslint-disable-next-line no-restricted-imports
+  MenuItem,
+  Text,
+  usePortal,
+} from '@sanity/ui'
+import {useCallback} from 'react'
+import {
+  ContextMenuButton,
+  getReleaseTone,
+  getVersionFromId,
+  type PublishDocumentVersionEvent,
+  RELEASES_INTENT,
+  Translate,
+  usePerspective,
+  useTranslation,
+  VersionInlineBadge,
+} from 'sanity'
+import {IntentLink} from 'sanity/router'
+import {usePaneRouter} from 'sanity/structure'
+
+import {MenuButton} from '../../../../../ui-components'
+import {structureLocaleNamespace} from '../../../../i18n'
+import {TIMELINE_MENU_PORTAL} from '../timelineMenu'
+
+export function PublishedEventMenu({event}: {event: PublishDocumentVersionEvent}) {
+  const {t} = useTranslation(structureLocaleNamespace)
+  const portalContext = usePortal()
+  const {params, setParams} = usePaneRouter()
+  const {setPerspective} = usePerspective()
+  const handleOpenReleaseDocument = useCallback(() => {
+    setParams({
+      ...params,
+      rev: event.versionRevisionId || '@lastPublished',
+      since: undefined,
+      historyEvent: event.id,
+      historyVersion: getVersionFromId(event.versionId),
+    })
+  }, [setParams, params, event])
+
+  const handleOpenDraftDocument = useCallback(() => {
+    setParams({
+      ...params,
+      rev: event.versionRevisionId,
+      preserveRev: 'true',
+      since: undefined,
+    })
+    setTimeout(() => {
+      // A bug is generated when we change the perspective and the params at the same time
+      // Resetting the params to the value it had before, because the paneRouter uses the previous value
```

Comment:
good catch, should this be something we add a story to revisit or is it fine to let it live?

Reply:
I think we can let it live, it will require a big change in how the pane router works